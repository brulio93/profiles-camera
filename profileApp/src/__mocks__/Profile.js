export default class Profile{
    name: string;
    phone: string;
    address: string;
    photo_uri: string;

    constructor(name, phone, address, photo_uri){
        this.name = name;
        this.phone = phone;
        this.address = address,
        this.photo_uri = photo_uri
    }
}