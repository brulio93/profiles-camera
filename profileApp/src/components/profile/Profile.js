import React from 'react';
import { View, Text, TextInput, Alert, Keyboard, AsyncStorage} from 'react-native';
import { Icon, Card, ListItem, FormInput, Button } from 'react-native-elements';
import styles from '../../styles/Styles';
import CameraComp from '../../components/camera-comp/CameraComp';
import { observer, inject } from 'mobx-react';

@inject('ProfileStore')
@observer
export default class Profile extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            name: '',
            phone: '',
            address: '',
            photo_uri: null,
            isCameraOn: false
        };
    }
    static navigationOptions = {
        title: null,
        headerBackTitle: null,
        header: null
     }; 


    componentDidMount(){
        const {ProfileStore} = this.props.ProfileStore;
        var global = this;

        AsyncStorage.getItem(ProfileStore.StorageUrl)
        .then((val) => { if(val){ global.props.ProfileStore.ProfileStore.profiles = JSON.parse(val); }} ).done();
    }
    createProfile(){
        const {navigation, ProfileStore} = this.props;
        try{
            ProfileStore.ProfileStore.addProfile({
                name: this.state.name,
                phone: this.state.phone,
                address: this.state.address,
                photo_uri: this.props.ProfileStore.ProfileStore.photo_uri
            });
        }
        catch(error){
            console.log(`Error: ${error}`);
        }
        navigation.goBack();
    }
    saveProfile(){
        if(this.state.name && this.state.phone && this.state.address){
            this.createProfile();            
            this.setState({
                name: '',
                phone: '',
                address: '',
                photo_uri: ''
            });           
            Alert.alert('Contact Saved');
        }
        else {
            Alert.alert('Fill all the fields');
        }
    }

    checkCamera(){
        if(this.state.isCameraOn){
            this.setState({
                isCameraOn: false,
            });
        } else {
            this.setState({
                isCameraOn: true,
            });
        }
    }
    
    renderCamera(){
        let content = null;
        if(this.state.isCameraOn){
            content = <Card title = 'Take a photo'>
                <View style={{height: 250}}>
                    <CameraComp toggleFunction = {this.checkCamera.bind(this)} />
                </View>
            </Card>
        }
        else {
            content = (            
                <Card title='Profile' image = { { uri: this.state.photo_uri || this.props.ProfileStore.ProfileStore.photo_uri } } >
                    <TextInput style={styles.textInput} placeholder="Name" onChangeText={(name) => this.setState({name})} value = {this.state.name} />
                    <TextInput style={styles.textInput} keyboardType='numeric' placeholder="Phone" onChangeText={(phone) => this.setState({phone})} value = {this.state.phone}/>
                    <TextInput style={styles.textInput} placeholder="Address" onChangeText={(address) => this.setState({address})} value = {this.state.address}/>
                </Card>
        );
        }
        return content; 
    }

    render(){
        let content = this.renderCamera();
        return(
            <View style={styles.indexContainer}>
                {content}                 
                <View style={{width: 300, padding: 10}}>
                    <Button style={styles.button} icon={{name: "camera", type: "font-awesome"}} backgroundColor="#00796B" onPress = {this.checkCamera.bind(this)} title="Change Picture"/>
                </View>                
                
                <View style={{width: 300, padding: 10}}>
                    <Button icon={{name: 'plus', type: 'font-awesome'}} backgroundColor="#517fa4" onPress = {this.saveProfile.bind(this)} title = 'Save Contact' />
                </View>               

            </View>
        );
    }
}