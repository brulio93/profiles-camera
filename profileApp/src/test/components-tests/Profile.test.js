import React from 'react';
import 'react-native';
import Profile from '../../components/profile/Profile';
import renderer from 'react-test-renderer';

const mockCamera = require('../../mocks/Camera').default;
const ProfileStore = require('../../stores/ProfileStore').default;

jest.mock('react-native-camera', () => mockCamera);

describe('Create a component', () => { 
  test('renders correctly', () => {
    const render = renderer.create(<Profile ProfileStore={ProfileStore.ProfileStore}/>).toJSON();
    expect(render).toMatchSnapshot();
  });
});
