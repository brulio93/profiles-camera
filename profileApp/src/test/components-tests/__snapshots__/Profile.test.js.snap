// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#e1e8ee",
          "borderWidth": 1,
          "margin": 15,
          "marginBottom": 0,
          "padding": 15,
          "shadowColor": "rgba(0,0,0, .2)",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "padding": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
          undefined,
        ]
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontSize": 17.5,
                  "fontWeight": "bold",
                  "marginBottom": 15,
                  "textAlign": "center",
                },
                Object {
                  "marginTop": 15,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          Profile
        </Text>
      </View>
      <View
        style={undefined}
      >
        <View
          style={
            Array [
              Object {
                "height": 150,
                "width": null,
              },
              undefined,
            ]
          }
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "https://octicons.github.com/img/og/mark-github.png",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "rgba(0, 0, 0, 0.2)",
                  "bottom": 0,
                  "flex": 1,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 22.5,
                      "fontWeight": "800",
                      "marginBottom": 8,
                    },
                    undefined,
                  ],
                ]
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16.25,
                      "fontWeight": "400",
                      "marginBottom": 8,
                    },
                    undefined,
                  ],
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "padding": 10,
              },
              undefined,
            ]
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Name"
            style={
              Object {
                "width": 280,
              }
            }
          />
          <TextInput
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="Phone"
            style={
              Object {
                "width": 280,
              }
            }
          />
          <TextInput
            onChangeText={[Function]}
            placeholder="Address"
            style={
              Object {
                "width": 280,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 10,
        "width": 300,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#517fa4",
              "marginBottom": 30,
              "width": 260,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#00796B",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 18,
                },
                Array [
                  Object {
                    "marginRight": 10,
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            type="font-awesome"
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Change Picture
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 10,
        "width": 300,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#517fa4",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 18,
                },
                Array [
                  Object {
                    "marginRight": 10,
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            type="font-awesome"
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Save Contact
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
